### PR TITLE
Simplification `intersection()` in `Ellipse` class

### PR DESCRIPTION
**Pull request by szymag**
I have cleaned up methods in `Ellipse` and `Circle` class. `intersection()` of two instances of `Circle` is done in `intersection()` in `Ellipse` class. There is no reason to consider this case separately. Unfortunately, I couldn't replace `solve` with `nonlinsolve` from `solveset`, because this method doesn't take a domain as a parameter, wherein we want our result. We solve system of a quadratic equation and  we may get complex results.

### CODE REVIEW

File: sympy/geometry/ellipse.py [hunk 1]
```diff
@@ -696,10 +693,7 @@ def intersection(self, o):
             # of intersection for coincidence first
             return self._do_line_intersection(o)
 
-        elif isinstance(o, Circle):
-            return self._do_ellipse_intersection(o)
-
-        elif isinstance(o, Ellipse):
+        elif isinstance(o, Circle) or isinstance(o, Ellipse):
```
Comment (smichr) [inline sympy/geometry/ellipse.py]: `elif isinstance(o, Ellipse):`  #since all circles are ellipses

**Comment by smichr** [comment]
I"m not sure (and I can't check it now) whether removal of the Circle intersection is the best idea. Sometimes such things are coded manually to give a simple result for a special case which might be more complicated if allowed to be solved "automatically" with solve. Do you think that's a possibility with what you removed?

**Comment by szymag** [comment]
I understand your concern. If something works, we shouldn't change that. Definitely, it isn't primary improvement. 
The possible result of intersection of two circles are: 
one point (circle are tangential)
two point (circle intersect each other, for ellipse, we can get also four intersect point)
zero point (circle are separated)
Ellipses can have also three common points - one tangential and two intersect.
Let's look up at the problem from equation side. We are dealing with system of two quadratic equations with two variables. General form of circle on plane are as follows:
(x-a)^2 + (y-b)^2 - r^2 == 0
I believe, that `solve` deals with such equation perfectly. Furthermore `solve` solves equations of ellipse which is a bit more complicated. We are also in real domain. 
Before, It was done similarly to solving that problem manually, on shit of paper.
I would also add, that every cases, that I mentioned is tested in line: `189`, `190`, `200`, `201`, `209`, `211`, `213`.

**Comment by smichr** [comment]
OK, as there is no test that demonstrates that the special handling is necessary then having less code to maintain is a positive direction. This is in.

**Pull request merged**